packages/test/test-version-utils/src/test/minCompatVersion.spec.ts

```diff
@@ -45,7 +107,9 @@ describe("Minimum Compat Version", () => {
 		);
 	});
 
-	for (let i = 1; i < numCompatVersions; i++) {
+	// N-1 check will fail while releasing a new version during bump version PR since pkgVersion will be one version ahead of latest release.
+	// In order to avoid conflicts every time we bump, we removed N-1 test.
```

Comment:
I don't recall the details of when we chatted about this (or something close), but to me the underlying problem is still that these unit tests depend on "the real world". `isCompatVersionBelowMinVersion()` below uses `getRequestedVersion()` which ends up potentially executing `npm` in a side process, and depending on what we have or have not published, the tests might or might not pass. That's a fundamental issue with the tests; I think we should figure out a way to refactor test-version-utils such that unit tests are really unit tests, deterministic.

cc @markfields , I think this might fit under tech debt.

Reply:
Do you suggest getting rid of all N-i tests then? I agree about unit tests need to be deterministic but we need a way to test "real world" scenarios since we use number compatConfig, not strings, and those numbers (-1..-9) will change every time we release.

Reply:
If the in-repo version has not been published, can you calculate and use the previous major or minor version as the repo version? Basically make the code act as though the repo version is the previous (before the bump) version. From my standpoint it seems like we want the code to behave the same pre and post bump.

I have similar concerns to @alexvy86  about relying on npm to enumerate versions. I don't know if it's better, but i have code in build tools that can enumerate versions based on the git tags in the repo. That said, I don't know if the version enumeration is the problem. Regardless of how you enumerate the versions, you are going to get the same list.

Reply:
Another way to think about it... the -1...-9 numbers describe offsets of a base version. How is that _base_ version calculated? I think if we can make that base version calculation handle the unpublished release case, then maybe everything else will "just work".

Reply:
That base version is calculated based on getCodeVersion() in case of CompatConfig being a number which I believe behaves similarly than pkgVersion. What we could do is add baseVersion as paramater to isCompatVersionBelowMinVersion and move from that fixed value instead of being calculated. That will handle the problem of non-deterministic tests as versions will be fixed. The problem will be that this fixed baseVersion will need to be manually updated every time we release. Maybe there is no way out of having to update these tests as we release by definition and by what are trying to test.

Reply:
In any case I believe fixing these tests shouldn't be release blocking. Maybe we can find a better pattern later and just go with current approach of just removing N-1 to unblock the bump PR.